jigglypuffRL/common/policies.py

```
@@ -8,28 +8,64 @@
 # TODO (ajaysub110): add other policy classes
 
 
-class MlpPolicy(nn.Module):
+class BasePolicy(nn.Module):
     """
-    Policy object that implements actor critic, using a MLP (2 layers of 24)
+    Base policy that all other policies should inherit.
     :param env: (Gym environment) The environment to learn from
-    :param n_hidden: (int) number of neurons in hidden layers
     """
 
-    def __init__(self, env, n_hidden=16):
-        super(MlpPolicy, self).__init__()
+    def __init__(self, env):
+        super(BasePolicy, self).__init__()
 
-        self.n_hidden = n_hidden
         self.state_space = None
         self.action_space = None
         self.env = env
 
         if isinstance(env.observation_space, gym.spaces.Box):
             self.state_space = env.observation_space.shape[0]
+        else:
+            print("Env type {} is not supported yet".format(type(env)))
 
         if isinstance(env.action_space, gym.spaces.Discrete):
             self.action_space = env.action_space.n
         elif isinstance(env.action_space, gym.spaces.Box):
             self.action_space = env.action_space.shape[0]
+        else:
+            print("Env type {} is not supported yet".format(type(env)))
+
+    def forward(self, x):
+        raise NotImplementedError
+
+    def sample_action(self, x):
+        x = self.forward(x)
+
+        if isinstance(self.env.action_space, gym.spaces.Discrete):
+            x = nn.Softmax(dim=-1)(x)
+            c = Categorical(probs=x)
+        elif isinstance(self.env.action_space, gym.spaces.Box):
+            mean, log_std = x
+            log_std = torch.clamp(log_std, -20, 2)
+            c = Normal(mean, log_std.exp())
+        else:
+            print(
+                "{} action spaces are not supported".format(type(self.env.action_space))
+            )
```

Again, we should probably raise an error here rather than printing